src/rest-api/class-base-api-controller.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/**
+ * Base API Controller
+ *
+ * @package Parsely
+ * @since   3.17.0
+ */
+
+declare(strict_types=1);
+
+namespace Parsely\REST_API;
+
+use Parsely\Parsely;
+
+/**
+ * Base API Controller.
+ *
+ * Used to define the namespace, version, and endpoints for an API controller. API controllers
+ * should extend this class and implement the `init` method to register endpoints.
+ *
+ * @since 3.17.0
+ */
+abstract class Base_API_Controller {
+
+	/**
+	 * The API namespace.
+	 *
+	 * @var string|false
+	 */
+	public const NAMESPACE = false;
+
+	/**
+	 * The API version.
+	 *
+	 * @var string
+	 */
+	public const VERSION = '';
+
+	/**
+	 * The route prefix, which acts as a namespace for the endpoints.
+	 *
+	 * @since 3.17.0
+	 *
+	 * @var string
+	 */
+	public const ROUTE_PREFIX = '';
+
+	/**
+	 * The endpoints.
+	 *
+	 * @var Base_Endpoint[]
+	 */
+	public $endpoints;
+
+	/**
+	 * The Parsely instance.
+	 *
+	 * @var Parsely
+	 */
+	public $parsely;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Parsely $parsely The Parsely instance.
+	 * @throws \UnexpectedValueException If the namespace is not defined.
+	 */
+	public function __construct( Parsely $parsely ) {
+		$this->parsely   = $parsely;
+		$this->endpoints = array();
+
+		if ( static::NAMESPACE === false ) {
+			throw new \UnexpectedValueException( 'The API controller must define a namespace.' );
```

Comment:
As a convention, let's use `use UnexpectedValueException` in the top of the file, and remove the slash here. Exactly as currently done in [src/rest-api/class-base-endpoint.php
](https://github.com/Parsely/wp-parsely/pull/2731/files#diff-cd68d5712fe1d6263916dad0e9ef0dc89f4bc367f02a85bb4d5e2ba0c7d4fa4e)